napari/layers/utils/layer_utils.py

```diff
@@ -204,81 +216,485 @@ def _nanmax(array):
+    # Vispy only supports uint8 RGB for which we set these clims
+    if rgb and dtype != np.uint8:
+        # Vispy does not display rgb other than uint8, so it casts it to uint8
+        warnings.warn(
+            trans._(
+                "RGB dtype is not np.uint8. Vispy will cast to np.uint8",
+                deferred=True,
+            )
+        )
+
+    if dtype == np.uint8 or rgb:
+        return 0, 255
+
+    shape = data.shape
+    chunk_size = _get_chunk_size(data)
+
+    # Iterable check is for TensorStore as this is required for the calculation, but TensorStore is not.
+    if (
+        chunk_size and np.prod(chunk_size) > PIXEL_THRESHOLD
+    ) or not isinstance(data, Iterable):
+        return None
+
+    if chunk_size:
+        shape = _get_blocks_grid_shape(data.shape, chunk_size)
+
+    if data.size > PIXEL_THRESHOLD and (data.ndim == 1):
+        slices = _get_1d_slices(shape, chunk_size)
         reduced_data = [
             [_nanmax(data[sl]) for sl in slices],
             [_nanmin(data[sl]) for sl in slices],
         ]
-    elif data.size > 1e7:
-        # If data is very large take the average of the top, bottom, and
-        # middle slices
+    elif data.size > PIXEL_THRESHOLD:
+        # If data is very large take the top, bottom, and middle slices
         offset = 2 + int(rgb)
-        bottom_plane_idx = (0,) * (data.ndim - offset)
-        middle_plane_idx = tuple(s // 2 for s in data.shape[:-offset])
-        top_plane_idx = tuple(s - 1 for s in data.shape[:-offset])
-        idxs = [bottom_plane_idx, middle_plane_idx, top_plane_idx]
-        # If each plane is also very large, look only at a subset of the image
-        if (
-            np.prod(data.shape[-offset:]) > 1e7
-            and data.shape[-offset] > 64
-            and data.shape[-offset + 1] > 64
-        ):
-            # Find a central patch of the image to take
-            center = [int(s // 2) for s in data.shape[-offset:]]
-            central_slice = tuple(slice(c - 31, c + 31) for c in center[:2])
-            reduced_data = [
-                [_nanmax(data[idx + central_slice]) for idx in idxs],
-                [_nanmin(data[idx + central_slice]) for idx in idxs],
-            ]
-        else:
-            reduced_data = [
-                [_nanmax(data[idx]) for idx in idxs],
-                [_nanmin(data[idx]) for idx in idxs],
-            ]
-        # compute everything in one go
-        reduced_data = dask.compute(*reduced_data)
+        # Indices are either numpy array or chunk indices dependent on data structure of data and are 0 < length <= 3
+        idxs = _get_plane_indices(shape, offset)
+        slices = _get_crop_slices(shape, idxs, offset, chunk_size)
+
+        reduced_data = [
+            [_nanmax(data[sl]) for sl in slices],
+            [_nanmin(data[sl]) for sl in slices],
+        ]
     else:
         reduced_data = data
 
+    if chunk_size:
+        reduced_data = dask.compute(*reduced_data)
```

Comment:
as far as I check calling das.compute is required only for dask arrays. The `zarr` looks not requiring this, so we may avoid import `dask` each time.